Lib/tempfile.py

```diff
@@ -868,10 +868,17 @@ def write(self, s):
         return rv
 
     def writelines(self, iterable):
-        file = self._file
-        rv = file.writelines(iterable)
-        self._check(file)
-        return rv
+        if self._max_size == 0 or self._rolled:
+            return self._file.writelines(iterable)
+
+        it = iter(iterable)
+
```

Comment:
Nitpick: remove this empty line if you apply my other suggestion. Empty lines inside a function separate semantic blocks of code, this is not a case.